src/hash/blake/mod.rs

```diff
@@ -276,10 +277,20 @@ where
     let digest = if Felt::IS_CANONICAL {
         blake3::hash(E::elements_as_bytes(elements))
     } else {
-        let mut hasher = blake3::Hasher::new();
-        for element in E::as_base_elements(elements) {
-            hasher.update(&element.as_int().to_le_bytes());
+        let blen = elements.len() << 3;
+
+        let mut bytes = Vec::with_capacity(blen);
+        #[allow(clippy::uninit_vec)]
+        unsafe {
+            bytes.set_len(blen)
+        }
+
+        for (idx, element) in E::as_base_elements(elements).iter().enumerate() {
+            bytes[idx * 8..(idx + 1) * 8].copy_from_slice(&element.as_int().to_le_bytes());
         }
+
+        let mut hasher = blake3::Hasher::new();
+        hasher.update(&bytes);
         hasher.finalize()
```

Comment:
I would replace with the following:
```Rust
let base_elements = E::as_base_elements(elements);
let num_bytes = base_elements.len() << 3;

let mut bytes = unsafe { uninit_vector(num_bytes) };
for (idx, element) in base_elements.iter().enumerate() {
    bytes[idx * 8..(idx + 1) * 8].copy_from_slice(&element.as_int().to_le_bytes());
}

blake3::hash(&bytes)
```
The main differences from your version are:
- We need to compute the number of bytes based on the number of base elements as extension elements could consist of multiple base elements.
- We can use `uniti_vector()` function we have in `miden-core` to create uninitialized vectors.

Reply:
> * We can use `uniti_vector()` function we have in `miden-core` to create uninitialized vectors.

Thanks, much better than fighting clippy.

Reply:
> * We need to compute the number of bytes based on the number of base elements as extension elements could consist of multiple base elements

Thanks, totally missed that.